### PR TITLE
[No Ticket][Risk=low] Srubenst/parameterize app header

### PR DESCRIPTION
**Pull request by blrubenstein**

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
 public class FireCloudConfig {
 
   public static final String X_APP_ID_HEADER = "X-App-ID";
```
Comment (gjuggler) [inline api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java]: Can you add a note describing our use of this header, and what expectations exist on the FireCloud side?

An alternative place to put this documentation / comment would be as a comment on the WorkbenchConfig class definition. I know we have zero comments there right now, but that's a pretty poor state of affairs & it's never too late to start leaving breadcrumbs of knowledge :)

File: api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java [hunk 2]
```diff
@@ -128,13 +127,13 @@ public StatusApi statusApi(WorkbenchConfig workbenchConfig) {
...
     apiClient.setDebugging(workbenchConfig.firecloud.debugEndpoints);
-    addFireCloudDefaultHeader(apiClient);
+    addFireCloudDefaultHeader(apiClient, workbenchConfig.firecloud.xAppIdValue);
     statusApi.setApiClient(apiClient);
     return statusApi;
   }
 
-  private void addFireCloudDefaultHeader(ApiClient apiClient) {
-    apiClient.addDefaultHeader(X_APP_ID_HEADER, X_APP_ID_HEADER_VALUE);
+  private void addFireCloudDefaultHeader(ApiClient apiClient, String value) {
```
Comment (gjuggler) [inline api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java]: This method feels a bit awkward, given it has only a single line and takes a parameter derived from the workbenchConfig, which this class already has access to.

Idea: maybe refactor this into a buildApiClient() method which handles as much common ApiClient logic as possible:

ApiClient apiClient = new ApiClient();
apiClient.setBasePath(workbenchConfig.firecloud.baseUrl);
apiClient.setDebugging(workbenchConfig.firecloud.debugEndpoints);
apiClient.addDefaultHeader(X_APP_ID_HEADER, workbenchConfig.firecloud.xAppIdValue);
return apiClient;

If you make that method public, we could even call it from FireCloudServiceImpl to remove a few lines of code from there too.

File: api/src/main/java/org/pmiops/workbench/notebooks/NotebooksConfig.java [hunk 3]
```diff
@@ -89,8 +89,8 @@ public ClusterApi serviceClusterApi(@Qualifier(NOTEBOOKS_SERVICE_CLIENT) ApiClie
     return api;
   }
 
-  private void addNotebooksDefaultHeader(ApiClient apiClient) {
+  private void addNotebooksDefaultHeader(ApiClient apiClient, String xAppHeaderValue) {
```
Comment (gjuggler) [inline api/src/main/java/org/pmiops/workbench/notebooks/NotebooksConfig.java]: Same comment here as above – maybe refactoring as an ApiClient builder method would make more sense?

If you don't think that path is worthwhile, I'd be inclined to remove this method and simply make the apiClient.addDeafultHeader call independently in each of the usage locations. This method feels a bit like needless redirection.

File: api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java (hunk 1 above)
Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java]: 👍 to this, adding it into WorkbenchConfig, and adding a tech debt ticket to do an audit of our config file and figure out what isn't clear so we can comment it.

Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java]: Filed: https://precisionmedicineinitiative.atlassian.net/secure/RapidBoard.jspa?rapidView=30&projectKey=RW&view=planning&selectedIssue=RW-2616&epics=visible&selectedEpic=RW-1801

**Comment by blrubenstein** [comment]
Codacy isn't analyzing for some reason. Going to merge since everything else is passing.

**Pull request merged**